### PR TITLE
[REEF-223]: Add a project reference to the JAR

### PR DESCRIPTION
**Pull request by beysims**
This addressed the issue by
Adding a project reference to JAR project.

JIRA: [REEF-223](https://issues.apache.org/jira/browse/REEF-223)

This closes #

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Driver/Bridge/ClrClientHelper.cs [hunk 1]
```diff
@@ -72,11 +71,34 @@ public static void Run(HashSet<string> appDlls, IConfiguration driverBridgeConfi
...
                 }
                 startInfo.RedirectStandardOutput = true;
+                startInfo.RedirectStandardError = true;
                 startInfo.UseShellExecute = false;
                 startInfo.CreateNoWindow = false;
                 LOGGER.Log(Level.Info, "Executing\r\n" + startInfo.FileName + "\r\n" + startInfo.Arguments);
+                LOGGER.Log(Level.Info, "Current Directroy " + Directory.GetCurrentDirectory());
                 using (Process process = Process.Start(startInfo))
                 {
+                    process.ErrorDataReceived += (sender, errorLine) =>
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Driver/Bridge/ClrClientHelper.cs]: This change is unrelated to the fix, right?

File: lang/cs/Org.Apache.REEF.Tests/Org.Apache.REEF.Tests.csproj [hunk 2]
```diff
@@ -78,11 +78,11 @@ under the License.
...
     <Compile Include="Utility\TestExceptions.cs" />
   </ItemGroup>
-  <ItemGroup>
-    <None Include="bin\run.cmd">
+  <ItemGroup>  
+    <None Include="run.cmd">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
-    <None Include="ConfigFiles\evaluator.conf">
+	<None Include="ConfigFiles\evaluator.conf">
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tests/Org.Apache.REEF.Tests.csproj]: Remove pure formatting changes, please.

File: lang/cs/Org.Apache.REEF.Tests/Org.Apache.REEF.Tests.csproj [hunk 3]
```diff
@@ -130,11 +130,6 @@ under the License.
     </ProjectReference>
   </ItemGroup>
   <ItemGroup>
-    <Content Include="bin\reef-bridge-java-0.11.0-incubating-SNAPSHOT-shaded.jar">
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tests/Org.Apache.REEF.Tests.csproj]: Is there a way to refer to this by the current version instead of hardcoding it?

File: lang/cs/Org.Apache.REEF.Client/Org.Apache.REEF.Client.csproj [hunk 4]
```diff
@@ -16,8 +16,7 @@ specific language governing permissions and limitations
 under the License.
 -->
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
-  <Import Project="$(SolutionDir)\build.props" />
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Client/Org.Apache.REEF.Client.csproj]: This looks like pure re-arranging of lines. Or did anything change here?

File: lang/cs/Org.Apache.REEF.Tests/Functional/ReefFunctionalTest.cs [hunk 5]
```diff
@@ -42,7 +42,7 @@ public class ReefFunctionalTest
         protected const string _stdout = "driver.stdout";
         protected const string _stderr = "driver.stderr";
         protected const string _cmdFile = "run.cmd";
-        protected const string _binFolder = "bin";
+        protected const string _binFolder = ".";
         // TODO: we will need a proper way to hide this when we are OSS'ed
         protected const string _blobStorageConnectionString =
             @"DefaultEndpointsProtocol=https;AccountName=reeftestlog;AccountKey=cuUmPRF9DiG56bciNc37O/SfHAoh1jFfJW6AsXAtWLPnjlOzXJGWgXhkyDFOGEHIMscqDU41ElUKnjcsJjWD9w==";
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tests/Functional/ReefFunctionalTest.cs]: This looks like a key in the code. Is this safe to have?

**Comment by markusweimer** [comment]
@beysims I am done with my pass.

File: lang/cs/Org.Apache.REEF.Tests/Org.Apache.REEF.Tests.csproj (hunk 3 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.Tests/Org.Apache.REEF.Tests.csproj]: This line was removed

File: lang/cs/Org.Apache.REEF.Client/Org.Apache.REEF.Client.csproj (hunk 4 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.Client/Org.Apache.REEF.Client.csproj]: It is re-arranging lines, but there is a semantics difference.

File: lang/cs/Org.Apache.REEF.Tests/Functional/ReefFunctionalTest.cs (hunk 5 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.Tests/Functional/ReefFunctionalTest.cs]: yes

File: lang/cs/Org.Apache.REEF.Tests/Org.Apache.REEF.Tests.csproj (hunk 3 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tests/Org.Apache.REEF.Tests.csproj]: D'oh :)

File: lang/cs/Org.Apache.REEF.Tests/Functional/ReefFunctionalTest.cs (hunk 5 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tests/Functional/ReefFunctionalTest.cs]: Why? Isn't this a key to a HDI instance?
